### PR TITLE
Refactor FXIOS-4573 [v105] history highlights adaptor

### PR DESCRIPTION
**Pull request by OrlaM**

### CODE REVIEW

File: Client/Frontend/Home/HistoryHighlights/HistoryHighlightsManager.swift [hunk 1]
```diff
@@ -64,11 +101,12 @@ class HistoryHighlightsManager {
     ///   - resultCount: The number of results to return
     ///   - completion: completion handler than contains either a list of `HistoryHighlights` if `shouldGroupHighlights` is set to false
     ///   or a combine list of `HistoryHighlights` and `ASGroup<HistoryHighlights>`if is true
-    public static func getHighlightsData(with profile: Profile,
```
Comment (lmarceau) [inline Client/Frontend/Home/HistoryHighlights/HistoryHighlightsManager.swift:67]: Yeah 🎉 no more static

File: Client/Frontend/Home/HistoryHighlights/HistoryHighlightsViewModel.swift [hunk 2]
```diff
@@ -248,25 +212,23 @@ extension HistoryHighlightsViewModel: HomepageViewModelProtocol, FeatureFlaggabl
...
+        if historyItems.count <= UX.maxNumberOfItemsPerColumn {
+            return historyItems.count
         } else {
             return numberOfColumns * UX.maxNumberOfItemsPerColumn
         }
     }
 
     var hasData: Bool {
-        return !(historyItems?.isEmpty ?? true)
+        return !(historyItems.isEmpty)
```
Comment (lmarceau) [inline Client/Frontend/Home/HistoryHighlights/HistoryHighlightsViewModel.swift:227]: nit
```suggestion
        return !historyItems.isEmpty
```

File: Client/Frontend/Home/HistoryHighlights/HistoryHighlightsViewModel.swift [hunk 3]
```diff
@@ -248,25 +212,23 @@ extension HistoryHighlightsViewModel: HomepageViewModelProtocol, FeatureFlaggabl
...
     }
 
     func updateData(completion: @escaping () -> Void) {
-        loadItems(completion: completion)
+        historyItems = historyHighlightsDataAdaptor.getHistoryHightlights()
     }
```
Comment (lmarceau) [inline Client/Frontend/Home/HistoryHighlights/HistoryHighlightsViewModel.swift:232]: Should we call the completion after the `historyItems =` ?

File: Client/Frontend/Home/HomepageViewModel.swift [hunk 4]
```diff
@@ -108,6 +108,11 @@ class HomepageViewModel: FeatureFlaggable {
 
         self.recentlySavedViewModel = RecentlySavedCellViewModel(
             profile: profile)
+        let historyDataAdaptor = HistoryHighlightsDataAdaptorImplementation(
+            profile: profile,
+            tabManager: tabManager,
+            urlBar: urlBar,
+            historyHighlightsDataAdaptor: historyDataAdaptor)
         self.historyHighlightsViewModel = HistoryHighlightsViewModel(
```
Comment (lmarceau) [inline Client/Frontend/Home/HomepageViewModel.swift:116]: Little issue here

Screen Shot 2022-08-11 at 5 42 16 PM

File: Tests/ClientTests/Frontend/Home/HistoryHighlights/HistoryHighlightsDataAdaptorTests.swift [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+        notificationCenter.notifiableListener = subject
+        self.subject = subject
+    }
+
+    override func tearDown() {
+        super.tearDown()
+        subject = nil
+        historyManager = nil
+        notificationCenter = nil
+    }
```
Comment (lmarceau) [inline Tests/ClientTests/Frontend/Home/HistoryHighlights/HistoryHighlightsDataAdaptorTests.swift:38]: Should be nullified too?
```suggestion
        notificationCenter = nil
        delegate = nil
    }
```

File: Tests/ClientTests/Frontend/Home/HistoryHighlights/HistoryHighlightsDataAdaptorTests.swift [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+    // Loads history on first launch with data
+    func testInitialLoadWithHistoryData() {
+        let item: HighlightItem = HistoryHighlight(score: 0, placeId: 0, url: "", title: "", previewImageUrl: "")
+        historyManager.callGetHighlightsDataCompletion(result: [item])
+
+        let results = subject.getHistoryHightlights()
+
+        XCTAssert(results.count == 1)
+        XCTAssert(historyManager.getHighlightsDataCallCount == 1)
+        XCTAssert(delegate.didLoadNewDataCallCount == 1)
```
Comment (lmarceau) [inline Tests/ClientTests/Frontend/Home/HistoryHighlights/HistoryHighlightsDataAdaptorTests.swift:49]: I would normally prefer XCTAssertEqual to check for equality 🤔

File: Tests/ClientTests/Frontend/Home/HistoryHighlights/HistoryHighlightsViewModelTests.swift [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+            score: 0,
+            placeId: 0,
+            url: "",
+            title: "mozilla",
+            previewImageUrl: "")
+        dataAdaptor.mockHistoryItems = [item1]
+
+        sut.didLoadNewData()
+
+        XCTAssertEqual(self.sut.getItemDetailsAt(index: 0)?.displayTitle, "mozilla")
```
Comment (lmarceau) [inline Tests/ClientTests/Frontend/Home/HistoryHighlights/HistoryHighlightsViewModelTests.swift:49]: nit
```suggestion
        sut.didLoadNewData()

        XCTAssertEqual(sut.getItemDetailsAt(index: 0)?.displayTitle, "mozilla")
```

File: Client/Frontend/Home/HistoryHighlights/HistoryHighlightsViewModel.swift (hunk 3 above)
Comment (lmarceau) [reply Client/Frontend/Home/HistoryHighlights/HistoryHighlightsViewModel.swift:232]: Mh, turns out I'll be removing completion in the clean up PR. You can disregard this comment

File: Client/Frontend/Home/HomepageViewModel.swift (hunk 4 above)
Comment (OrlaM) [reply Client/Frontend/Home/HomepageViewModel.swift:116]: Oops, had a merge conflict and things got jumbled but I never actually committed the fix I put in for it 🙈

**Review by lmarceau — APPROVED** [review]

**Pull request merged**